native_client/deepspeech.cc

```diff
@@ -8,45 +8,423 @@
 #endif
 
 #include <iostream>
+#include <vector>
 
 #include "deepspeech.h"
-#include "deepspeech_utils.h"
 #include "alphabet.h"
 #include "beam_search.h"
 
 #include "tensorflow/core/public/session.h"
 #include "tensorflow/core/platform/env.h"
 #include "tensorflow/core/util/memmapped_file_system.h"
 
-#define BATCH_SIZE 1
+#include "c_speech_features.h"
+
+//TODO: infer batch size from model/use dynamic batch size
+const int BATCH_SIZE = 1;
+
+//TODO: use dynamic sample rate
+const int SAMPLE_RATE = 16000;
+
+//TODO: infer n_steps from model
+const int N_STEPS_PER_BATCH = 16;
+
+const float AUDIO_WIN_LEN = 0.025f;
+const float AUDIO_WIN_STEP = 0.01f;
+const int AUDIO_WIN_LEN_SAMPLES = (int)(AUDIO_WIN_LEN * SAMPLE_RATE);
+const int AUDIO_WIN_STEP_SAMPLES = (int)(AUDIO_WIN_STEP * SAMPLE_RATE);
+
+const int MFCC_FEATURES = 26;
+const int MFCC_CONTEXT = 9;
+const int MFCC_WIN_LEN = 2 * MFCC_CONTEXT + 1;
+const int MFCC_FEATS_PER_TIMESTEP = MFCC_FEATURES * MFCC_WIN_LEN;
+
+const float COEFF = 0.97f;
+const int N_FFT = 512;
+const int N_FILTERS = 26;
+const int LOWFREQ = 0;
+const int CEP_LIFTER = 22;
 
 using namespace tensorflow;
 using tensorflow::ctc::CTCBeamSearchDecoder;
 using tensorflow::ctc::CTCDecoder;
 
 namespace DeepSpeech {
 
+class StreamingState {
```

Comment:
As much as I can see, this class is completely defined here. In `deepspeech.h` we only have a `class StreamingState;` statement. Therefore, I think it might be useful that this is being properly documented, since the `StreamingState` is exposed to the public.

Reply:
It's an opaque pointer, so the layout is not exposed to API consumers. Is the documentation in setupStream/feedAudioContent/finishStream not enough?

Reply:
You have a point here. The other docs are good, I missed the fact that since it's not in the header it's not publicly exposed. We got some people on Discourse asking to play with the logits, if I remember correctly. Maybe there's room to expose some of that?